repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector.java

```diff
@@ -138,31 +135,38 @@ public String getRemoteBranchName() {
     public void run() {
         if (this.buildToBeInspected.getStatus() != ScannedBuildStatus.PASSING_AND_PASSING_WITH_TEST_CHANGES) {
             AbstractStep cloneRepo = new CloneRepository(this);
-            cloneRepo.setNextStep(new CheckoutBuggyBuild(this))
-                    .setNextStep(new ComputeSourceDir(this, true))
-                    .setNextStep(new ComputeTestDir(this))
+            AbstractStep lastStep = cloneRepo
+                    .setNextStep(new CheckoutBuggyBuild(this, true))
+                    .setNextStep(new ComputeSourceDir(this, true, true)) // TODO: check, should it be really blocking?
+                    .setNextStep(new ComputeTestDir(this, true))                    // IDEM
                     .setNextStep(new ResolveDependency(this))
                     .setNextStep(new BuildProject(this))
                     .setNextStep(new TestProject(this))
-                    .setNextStep(new GatherTestInformation(this, new BuildShouldFail(), false))
+                    .setNextStep(new GatherTestInformation(this, true, new BuildShouldFail(), false))
                     .setNextStep(new InitRepoToPush(this))
                     .setNextStep(new PushIncriminatedBuild(this))
-                    .setNextStep(new NPERepair(this))
-                    .setNextStep(new ComputeClasspath(this))
-                    .setNextStep(new ComputeSourceDir(this, false))
-                    .setNextStep(new AssertFixerRepair(this))
-                    .setNextStep(new AstorRepair(this))
-                    .setNextStep(new NopolRepair(this))
-                    .setNextStep(new CommitPatch(this, false))
-                    .setNextStep(new CheckoutPatchedBuild(this))
+                    .setNextStep(new ComputeClasspath(this, false))
+                    .setNextStep(new ComputeSourceDir(this, false, false));
+
+            for (String repairToolName : RepairnatorConfig.getInstance().getRepairTools()) {
+                AbstractRepairStep repairStep = RepairToolsManager.getStepFromName(repairToolName);
+                if (repairStep != null) {
+                    repairStep.setProjectInspector(this);
+                    lastStep = lastStep.setNextStep(repairStep);
+                } else {
+                    logger.error("Error while getting step class for following name: " + repairToolName);
+                }
+            }
+
+            lastStep.setNextStep(new CommitPatch(this, false))
+                    .setNextStep(new CheckoutPatchedBuild(this, true))
                     .setNextStep(new BuildProject(this))
                     .setNextStep(new TestProject(this))
-                    .setNextStep(new GatherTestInformation(this, new BuildShouldPass(), true))
+                    .setNextStep(new GatherTestInformation(this, false, new BuildShouldPass(), true))
```

Comment:
Just to check if the `false` here is correct.

Reply:
Hmmm actually I'm not sure, I think I'll change that.